shared/src/main/scala/ListHelpers.scala

```diff
@@ -104,8 +106,42 @@ object ListHelpers:
   end mold
 
   def sortBy(iterable: VList, key: VFun)(using ctx: Context): VList =
-    val branches = key.originalAST match
-      case Some(lam) => lam.body
+    key.originalAST match
+      case Some(lam) =>
+        val branches = lam.body
+        if branches.length < 2 then
+          return VList(
+            iterable.zipWithIndex
+              .sorted((a, b) =>
+                MiscHelpers.compareExact(
+                  key.executeResult(a(0), a(1), List(a(0))),
+                  key.executeResult(b(0), b(1), List(b(0)))
+                )
+              )
+              .map(_._1)*
+          )
+
+        val out = iterable.zipWithIndex
+          .sortWith { (a, b) =>
+            val (aRes, bRes) =
+              branches.view
+                .map { branch =>
+                  val f =
+                    VFun.fromLambda(AST.Lambda(1, List.empty, List(branch)))
+                  (
+                    f.execute(a(0), a(1), List(a(0))),
+                    f.execute(b(0), b(1), List(b(0)))
+                  )
+                }
+                .find(_ != _)
+                .getOrElse(
+                  (ctx.settings.defaultValue, ctx.settings.defaultValue)
+                )
+            MiscHelpers.compareExact(aRes, bRes) < 0
```

Comment:
```suggestion
            branches.view
                .map { branch =>
                  val f =
                    VFun.fromLambda(AST.Lambda(1, List.empty, List(branch)))
                  (
                    f.execute(a(0), a(1), List(a(0))),
                    f.execute(b(0), b(1), List(b(0)))
                  )
                }
                .find(_ != _)
                .map { case (aRes, bRes) => MiscHelpers.compareExact(aRes, bRes) < 0 }
                .getOrElse(false)
```
Didn't remove the `val ... =` part before